src/bn_txns.erl

```diff
@@ -45,7 +50,29 @@ init(Args) ->
             init(Args);
         {ok, State} ->
             persistent_term:put(?MODULE, State),
-            {ok, State#state{}}
+            case application:get_env(blockchain, store_historic_balances, false) of
+                true ->
+                    BalancesDir = filename:join(proplists:get_value(base_dir, Args, "data"), ?BALANCES_DB_FILE),
+                    ok = filelib:ensure_dir(BalancesDir),
+                    GlobalOpts = application:get_env(rocksdb, global_opts, []),
+                    DBOptions = [{create_if_missing, true}, {atomic_flush, true}] ++ GlobalOpts,
+                    case rocksdb:open(BalancesDir, DBOptions) of
+                        {ok, BalancesDB} ->
+                            ChainHeightTab = ets:new(chain_height_tab, []),
```

Comment:
You shouldn't use an ETS table here.. Store the balance height in a special keyed entry in the rocks db table for balances